keras/backend/mxnet_backend.py

```diff
@@ -3043,8 +3043,11 @@ def conv1d(x, kernel, strides=1, padding='valid',
 
     # Causal requires temporal padding.
     # MXNet backend does not support temporal padding on 3D tensor.
-    if padding is 'causal':
-        raise ValueError('MXNet Backend: conv1d does not support "causal" padding mode')
+    kernel_shape = kernel.shape
+    if padding == 'causal':
+        pad = dilation_rate * (kernel_shape[0] - 1)
```

Comment:
Can you please code comments describing the logic. Will help us later.

Reply:
done